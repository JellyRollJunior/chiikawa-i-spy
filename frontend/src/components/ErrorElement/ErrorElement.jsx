import { IconWrapper } from '../IconWrapper/IconWrapper.jsx';
import styles from './ErrorElement.module.css';
import shared from '../../styles/shared.module.css';
import kurimanju from '../../assets/error/kurimanju.png';
import usagi from '../../assets/error/usagi.png';
import chiikawa from '../../assets/error/chiikawa.png';

const messages = [
  {
    src: kurimanju,
    author: 'Ancient wise badger thing',
    quote: 'I need a beer',
    icons: 2,
  },
  {
    src: usagi,
    author: 'Ancient wise rabbit',
    quote: 'Yaha Yaha unma unma Yaha',
    icons: 5,
  },
  {
    src: chiikawa,
    author: 'Ancient wise hamder tingy',
    quote: 'u u wa wa uwa',
    icons: 0,
  },
];

const ErrorElement = ({ error, msgId = 0, children }) => {
  if (msgId >= messages.length) msgId = 0;

  return (
    <section
      className={`${shared.card} ${styles.errorWrapper} ${shared.marginTopMedium}`}
    >
      <IconWrapper
        size={36}
        margin={12}
        leftIcon={messages[msgId].icons}
        rightIcon={messages[msgId].icons}
      >
        <h2>{error}</h2>
      </IconWrapper>
      <img
        src={messages[msgId].src}
        alt="kurimanju looking at you"
        className={`${styles.errorUsagi} ${shared.marginTopSmall}`}
      />
      <p>
        {messages[msgId].author} once said:
        <br />
        {messages[msgId].quote}
      </p>
      {children}
    </section>
  );
};

export { ErrorElement };
