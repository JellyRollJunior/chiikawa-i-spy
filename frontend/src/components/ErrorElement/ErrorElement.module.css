.errorUsagi {
  height: 175px;
}

.errorWrapper {
  width: fit-content;
  padding: 12px 24px;
  color: var(--dark-green);
  text-align: center;
}
