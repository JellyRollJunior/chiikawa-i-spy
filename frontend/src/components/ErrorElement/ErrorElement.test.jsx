import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { ErrorElement } from './ErrorElement.jsx';

describe('Error element component', () => {
  it('renders title, character image, quote, and children', () => {
    const { container } = render(
      <ErrorElement error={'error!'} msgId={0}>
        <button>I am a child</button>
      </ErrorElement>
    );

    expect(container).toMatchSnapshot();
  });
});
