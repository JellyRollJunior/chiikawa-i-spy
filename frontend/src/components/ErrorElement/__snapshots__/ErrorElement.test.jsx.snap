// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error element component > renders title, character image, quote, and children 1`] = `
<div>
  <section
    class="_card_5890fc _errorWrapper_ef92cb _marginTopMedium_5890fc"
  >
    <div
      class="_iconWrapper_5d4a96"
    >
      <img
        alt="Chiikawa character icon"
        src="/src/assets/characterIcons/kurimanju-icon.png"
        style="height: 36px; width: 36px; margin-right: 12px;"
      />
      <h2>
        error!
      </h2>
      <img
        alt="Chiikawa character icon"
        src="/src/assets/characterIcons/kurimanju-icon.png"
        style="height: 36px; width: 36px; margin-left: 12px;"
      />
    </div>
    <img
      alt="Ancient wise badger thing"
      class="_errorUsagi_ef92cb _marginTopSmall_5890fc"
      src="/src/assets/error/kurimanju.png"
    />
    <p>
      Ancient wise badger thing
       once said:
      <br />
      I need a beer
    </p>
    <button>
      I am a child
    </button>
  </section>
</div>
`;
